From c66c2fb87228dae90ae4a802846844890feb332f Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Mon, 27 Jul 2026 17:48:43 +0800
Subject: [PATCH] mtk: hostapd: fix heap buffer overflow in
 hostapd_parse_argument_helper()

fixup! mtk: hostapd: add connac3 PHY MURU manual mode config support

Bound the parsing loop to MAX_MU_CTRL_NUM so input[cnt++] cannot write
past the 17-element buffer when given more colon-separated values.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 hostapd/ctrl_iface.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 3290c13ef..2fd056b41 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -4911,8 +4911,8 @@ hostapd_parse_argument_helper(char *value, u16 **ptr_input)
 		wpa_printf(MSG_ERROR, "Failed to allocate memory.\n");
 		return -1;
 	}
-	while (value) {
-		u8 val = strtol(value, &endptr, 10);
+	while (value && cnt < MAX_MU_CTRL_NUM) {
+		u16 val = strtol(value, &endptr, 10);
 
 		if (value != endptr) {
 			input[cnt++] = val;
-- 
2.45.2

